tensorrt_llm/_torch/pyexecutor/_util.py

```diff
@@ -455,12 +456,25 @@ def create_py_executor_instance(
 
         num_experts = _try_infer_num_experts(model_engine.model.model_config)
 
+        num_attn_layers = model_binding_config.num_attention_layers()
+        per_layer_kv_heads = [
+            model_binding_config.num_kv_heads(i) for i in range(num_attn_layers)
+        ]
+        num_kv_attention_heads = per_layer_kv_heads[0] if len(
+            set(per_layer_kv_heads)) == 1 else max(per_layer_kv_heads)
+        if len(set(per_layer_kv_heads)) > 1:
+            # NOTE: This code-path is currently untested and not validated. Can fail!
+            # This support is tracked in TRTLLM-6561
+            logger.warning(
+                f"Non-uniform KV heads per layer detected, using max ({num_kv_attention_heads}) for LoRA. This code-path is currently untested and not validated. May fail!"
+            )
```

Comment:
adding tests and validating/verifying this was beyong the scope of this PR. Have created the ticket to track this - but for now, as part of the changes need to pass-through the kv_heads, this was a side-effect.

CC @Naveassaf @tijyojwad @shaharmor98 @amitz-nv

Reply:
that's reasonable, thanks @venkywonka